### ISSUE TITLE
Viewer documentation assumes interactive terminal

### ISSUE THREAD

**Issue by stefanv** [napari/napari#3922]
The [viewer documentation](https://napari.org/tutorials/fundamentals/viewer.html) assumes you are not writing code in a Python file, and therefore does not add `.show()` to the napari viewer. Some text explaining that this is necessary outside an interactive, matplotlib-aware context would be helpful.

**Comment by tlambert03** [comment]
Thanks @stefanv, very good point.

the API is subtly different from matplotlib here though (and nothing is matplotlib-aware here, just `%gui qt` from IPython). Technically the viewer _widget_ is "visible" by default (what we need here is the event loop if we're outside of IPython).   So, to the docs, we would add `napari.run()` to the end of teh script, as shown in the majority of our [examples](https://github.com/napari/napari/tree/main/examples)

(or, if you want to block the progression of the script), `viewer.show(block=True)` should work too.

anyway, definitely should be made clearer in the docs.

**Comment by stefanv** [comment]
Yes, of course---event loop, not matplotlib, sorry!  And `block=True` was what I was going for.
